torch_geometric/explain/algorithm/gnn_explainer.py

```diff
@@ -0,0 +1,317 @@
+            settings in
+            :attr:`~torch_geometric.explain.algorithm.GNNExplainer.coeffs`.
+    """
+
+    coeffs = {
+        'edge_size': 0.005,
+        'edge_reduction': 'sum',
+        'node_feat_size': 1.0,
+        'node_feat_reduction': 'mean',
+        'edge_ent': 1.0,
+        'node_feat_ent': 0.1,
+    }
+
+    def __init__(
+        self,
+        epochs: int = 100,
+        lr: float = 0.01,
+        shared_mask: bool = True,
+        **kwargs,
+    ):
+        super().__init__()
+        self.epochs = epochs
+        self.lr = lr
+        self.coeffs.update(kwargs)
+        self.shared_mask = shared_mask
+        self.node_mask = None
+        self.edge_mask = None
+
+    def forward(self, model: torch.nn.Module, x: Tensor, edge_index: Tensor,
+                explainer_config: ExplainerConfig, model_config: ModelConfig,
+                target: Tensor, target_index: Optional[Union[int,
+                                                             Tensor]] = None,
+                index: Optional[int] = None, **kwargs) -> Explanation:
+
+        if isinstance(target_index, Tensor):
+            raise ValueError(
+                'GNNExplainer only supports single target index for now')
+
+        # can it be an issue for some models ?
+        model.eval()
+        self.clear(model)
+
+        if model_config.task_level == ModelTaskLevel.node:
+            node_mask, edge_mask = self.explain_node(model, x, edge_index,
+                                                     explainer_config,
+                                                     model_config, target,
+                                                     index, target_index,
+                                                     **kwargs)
+        elif model_config.task_level == ModelTaskLevel.graph:
+            node_mask, edge_mask = self.explain_graph(model, x, edge_index,
+                                                      explainer_config,
+                                                      model_config, target,
+                                                      target_index, index=None,
+                                                      **kwargs)
+        else:
+            raise ValueError(
+                'GNNExplainer does not support task level {}'.format(
+                    model_config.task_level))
+        if explainer_config.node_mask_type == MaskType.attributes:
+            if self.shared_mask:
+                node_feat_mask = torch.stack([node_mask] * x.size(0), dim=0)
+            else:
+                node_feat_mask = node_mask
+            node_mask = None
+        else:
+            node_feat_mask = None
+        self.clear(model)
+
+        # build explanation
+        return Explanation(x=x, edge_index=edge_index, edge_mask=edge_mask,
+                           node_mask=node_mask, node_feat_mask=node_feat_mask)
+
+    def explain_graph(self, model: torch.nn.Module, x: Tensor,
```

Comment:
```suggestion
    def _explain_graph(self, model: torch.nn.Module, x: Tensor,
```
Nitpick: lets add `_` to methods that are not used outside this class.